### DISCUSSION TITLE
clarify licensing model before upload to Github

### DISCUSSION THREAD

**Discussion by urapadmin** [arch-kiosk/arch-kiosk-office#1244] [General] [answered]
This month I plan to start uploading source code to Github. From that moment on we need a license.
On a quick glance I think I would prefer the Gnu GPLv3 over MIT because we don't want others to distribute a version of our project as closed source. But a quick glance is not necessarily the right thing. Here is what I think we need to do:

- [ ] look at all the open source libraries that are part of our software (a number in the hundreds) to find out if there is a license that limits our licensing. E.g. if there is a GPLv3 in one of the libraries, that would presumably keep us from making our own software MIT.
- [ ] ask somebody at Brown what to do and if there are guidelines.

**Comment by lbestock** [comment]
Somebody at Brown is my side, but where? My instinct is to contact the Computer Science department, which is a connection I would like to have anyway. Another option is the Center for Digital Scholarship but I don't trust them to know as much - I don't think they often deal with the creation of software. I also don't trust them to say "I don't know" if they don't know. Nelson Center for Entrepreneurship is another possibility and another connection I would like.

**Comment by urapadmin** [comment]
you could look at other Brown Github repositories first and maybe get in contact with one of those teams

**Comment by lbestock** [comment]
https://ithelp.brown.edu/kb/articles/github-for-brown

**Comment by urapadmin** [comment]
see #996

**Comment by lbestock** [comment]
I have sent an inquiry to the very top of the Center for Digital Scholarship to ask to whom we should speak. He won't be helpful himself but might know who can, and it was strategic to get the issue to his attention. But he's out of office.

**Comment by PeterDeathBredon** [comment]
There is a GNU Affero 3 license that fills a gap of the GNU 3 licensing model: https://www.gnu.org/licenses/agpl-3.0.de.html. Looks better.

**Comment by urapadmin** [comment]
some discussions should come to an end ...

**Comment by urapadmin** [comment]
also: https://opensource.org/licenses/AGPL-3.0. It is a long text. But the preamble shows that this is indeed relevant for us particularly should we move towards what the kids call "headless" architecture, where the server just serves an api but not a ui ani more. 

But a professional could fork this and sell it for money as far as I understand (they would have to disclose the source doe, though). We want to allow that, right?

**Comment by urapadmin** [comment]
node.js (THE core installation management tool for javascript development) suggests what seems to be popular in the javascript world: ISC. But that seems to be just carte blanche to me.

**Comment by urapadmin** [comment]
all licenses seem to want an "author". Who would that be? Can it be Brown? Should it be me? Both?

**Comment by urapadmin** [comment, marked as answer]
for the time being we settle on AGPL3

**Comment by lbestock** [comment]
Questions that have arisen and need @lbestock to research: do we need an author at all as a requirement of the license? Do we need an author for someone else to cite if they use the software?

**Comment by lbestock** [comment]
A few quotes and notes coming out of the FAQ on GPP, which does not, alas, address issues of authorship by that name.

"The GPL requires all copies to carry an appropriate copyright notice." - is this relevant to authorship?
But it is not permitted to include a requirement for citation or acknowledgement in research papers.

"Just putting a copy of the GNU GPL in a file in your repository does not explicitly state that the code in the same repository may be used under the GNU GPL. Without such a statement, it's not entirely clear that the permissions in the license really apply to any particular source file. An explicit statement saying that eliminates all doubt.

A file containing just a license, without a statement that certain other files are covered by that license, resembles a file containing just a subroutine which is never called from anywhere else. The resemblance is not perfect: lawyers and courts might apply common sense and conclude that you must have put the copy of the GNU GPL there because you wanted to license the code that way. Or they might not. Why leave an uncertainty?

This statement should be in each source file. A clear statement in the program's README file is legally sufficient as long as that accompanies the code, but it is easy for them to get separated."

"You should put a notice at the start of each source file, stating what license it carries, in order to avoid risk of the code's getting disconnected from its license. If your repository's README says that source file is under the GNU GPL, what happens if someone copies that file to another program? That other context may not show what the file's license is."

Is there a difference between copyright holder and author? A university can be a copyright holder.

Hm. I think I would prefer you have copyright on this, and from a time when you are NOT working at Brown, so that there is no possibility later that they can try to make it proprietary and charge for it.

**Reply by urapadmin** [reply]
>Hm. I think I would prefer you have copyright on this, and from a time when you are NOT working at Brown, so that there is no possibility later that they can try to make it proprietary and charge for it.

As I understand this, should somebody - no matter who - decide to make it proprietary, the license would bind them nonetheless to make that version open source. The license is exactly there for that reason: You are not allowed to take the source code and make a project on its basis that is closed source.
